### PR TITLE
txn: add rollback reason for check_txn_status

### PR DESCRIPTION
**Pull request by youjiali1995**
Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

###  What have you changed?
Add rollback reason for `kv_check_txn_status` to investigate why unit test fails.

TiDB https://github.com/pingcap/tidb/pull/13432

###  What is the type of the changes?
- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?
Please select the tests that you ran to verify your changes:
- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No

###  Does this PR affect `tidb-ansible`?
No

### CODE REVIEW

File: src/storage/txn/process.rs [hunk 1]
```diff
@@ -841,7 +841,7 @@ fn process_write_impl<S: Snapshot, L: LockMgr>(
             // The lock is possibly resolved here only when the `check_txn_status` cleaned up the
             // lock, and this may happen only when it returns `Rollbacked`.
```
Comment (sticnarf) [inline src/storage/txn/process.rs]: The comment should be updated.

File: src/storage/types.rs [hunk 2]
```diff
@@ -69,10 +69,12 @@ impl Mutation {
 /// Represents the status of a transaction.
 #[derive(PartialEq, Debug)]
 pub enum TxnStatus {
-    /// The txn is just rolled back by the current command.
-    Rollbacked,
     /// The txn was already rolled back before.
     RollbackedBefore,
```
Comment (MyonKeminta) [inline src/storage/types.rs]: How about just name this `Rollbacked` here? When I wrote this I can't come up with a better name, and have to distinguish it from the old one `Rollbacked`, so I named it `RollbackedBefore`.

Comment (youjiali1995) [reply src/storage/types.rs]: Good idea.

**Comment by youjiali1995** [comment]
@sticnarf PTAL again. Thanks!

**Pull request merged**